### PR TITLE
Add craftable dry batteries (zinc-carbon dry cell)

### PR DESCRIPTION
**Pull request by Proxima-b**
#### Summary
Content "Adds new item 'Dry battery'"

#### Purpose of change
The disposible alkaline batteries currently requires material from the labs to craft which is not always an option early game, especially in a wildness surviving gamethrough with innawood or rural_mapgen mods.

#### Describe the solution
This addition adds dry batteries which do not require lab materials to craft, instead, using material which can be mass produced by the players. However this kind of batteris have much lower energy densities than disposable alkaline batteries.

#### Describe alternatives you've considered
Nothing of sort.

#### Testing
Playtested, crafted and used.

#### Additional context
https://www.epectec.com/batteries/chemistry/
This article articulates the difference of energy density between Zinc–carbon primary batteries and Zinc-MnO2 Alkaline batteries, which is roughly consistant with the number I have set, based on the current data in the game.

### CODE REVIEW

File: data/json/recipes/other/power_supplies.json [hunk 1]
```diff
@@ -613,6 +633,26 @@
...
+    "type": "recipe",
+    "activity_level": "MODERATE_EXERCISE",
+    "result": "light_dry_cell",
+    "category": "CC_ELECTRONIC",
+    "subcategory": "CSC_ELECTRONIC_TOOLS",
+    "skill_used": "chemistry",
+    "skills_required": [ "electronics", 2 ],
+    "difficulty": 2,
+    "time": "1 h",
+	"autolearn": true,
```
Comment (BrettDong) [inline data/json/recipes/other/power_supplies.json:646]: Needs formatting.

Comment (Proxima-b) [reply data/json/recipes/other/power_supplies.json:646]: Please elaborate, the formating seems right to me... I do not understand.

Comment (Proxima-b) [reply data/json/recipes/other/power_supplies.json]: resolved

**Comment by Proxima-b** [comment]
Why is there a json style check error and how to fix it?

**Comment by NetSysFire** [comment]
Click on the details to get the output of the test, which tells you what needs to be fixed. There is also `tools/format/json_formatter.cgi` which can do that automagically for you.
You also need to fix the summary of your PR.

**Comment by NetSysFire** [comment]
Please do read https://github.com/CleverRaven/Cataclysm-DDA/blob/master/.github/CONTRIBUTING.md#all-prs-should-have-a-summary-section-with-one-line

**Comment by Proxima-b** [comment]
> Please do read https://github.com/CleverRaven/Cataclysm-DDA/blob/master/.github/CONTRIBUTING.md#all-prs-should-have-a-summary-section-with-one-line

Thank you, I have fixed it accordingly.

**Comment by Maddremor** [comment]
Can you give an example of how one would make a crude battery? Not that I doubt the chemistry, but a practical example (even if only a school science project) could be useful for not skipping a step by accident.

**Comment by Proxima-b** [comment]
> Can you give an example of how one would make a crude battery? Not that I doubt the chemistry, but a practical example (even if only a school science project) could be useful for not skipping a step by accident.

That I actully have no idea, but there are already recipes for alkaline batteries, I guess the methods used in making dry batteries can't be that different.
But If I have to guess, I think it would be:
-Smelt a casing somehow.
-Place anode and cathod materials and insulation into the casing with as much surface area as possible.(without touching eachother of course)
-Stuff the remaning space with electrolytes (any state).
-Seal the casing.

**Comment by BrettDong** [comment]
> The hosted runner: GitHub Actions 4 lost communication with the server. Anything in your workflow that terminates the runner process, starves it for CPU/Memory, or blocks its network access can cause this error.

**Comment by Proxima-b** [comment]
> > The hosted runner: GitHub Actions 4 lost communication with the server. Anything in your workflow that terminates the runner process, starves it for CPU/Memory, or blocks its network access can cause this error.

But my content seems ok , I just don't understand how to resolve this

**Comment by kevingranade** [comment]
Good news, detailed source
https://www.doitpoms.ac.uk/tlplib/batteries/batteries_zn_c.php
Bad news, this is much more complicated than you seem to think. I don't see this being a workable innawoods recipe. It needs a sintered carbon rod and a complex casing.

In broad strokes, I don't think there's an intersection between portable and innawoods craftable, though I'm open to reviewing sources if you find any candidates.

A voltaic pile as you originally outlined would work, but is low voltage and I'm guessing has extremely low energy density.
A lead-acid battery is a good option but needs a difficult to produce casing and isn't portable.

**Comment by Alex-Folts** [comment]
Carbon rod is not a problem, it would require extra work to make it.
This process will do: https://www.youtube.com/watch?v=N9B2OAiu95Y

Main problem of this sort of batteries is getting zinc metal.

**Comment by kevingranade** [comment]
This also highlights that the existing craftable cells need to be revisited, the casing is pretty complex and there's no electrode in the recipe.  Ideally you could harvest battery casings and replace/refresh the active ingredients, after looking at their structure I'm increasingly skeptical of from scratch cells that are drop-in replacements for modern batteries.

Thanks for the electrode source, that seems feasible for a battery electrode at least, though I have my doubts for other applications.

**Comment by Proxima-b** [comment]
> This also highlights that the existing craftable cells need to be revisited, the casing is pretty complex and there's no electrode in the recipe. Ideally you could harvest battery casings and replace/refresh the active ingredients, after looking at their structure I'm increasingly skeptical of from scratch cells that are drop-in replacements for modern batteries.
> 
> Thanks for the electrode source, that seems feasible for a battery electrode at least, though I have my doubts for other applications.

The crafting requires blacksmith standard tools with smelting, which is exactly how a complex metal process procedure is represented in dda.
If we look closer, many crafting process of the electronic devices in the game are somewhat simplified, like " electrohackl" and "handheld game system", or the UPS,  which I'm pretty sure need a dust-free workshop or something else to craft.
For lack of a complete industrial system, I think these somewhat simplified recipes do well to balance the gameplay and realism elements of the game.
I think the dry battery pieces do not break the overall immersion and game balance. As for the problem of existing craftable cells, they do produce pre-apoclype batteries, so I think add the requirement of vaccum molder and polycarbonate sheets would suffice.

**Comment by Proxima-b** [comment]
> Click on the details to get the output of the test, which tells you what needs to be fixed. There is also `tools/format/json_formatter.cgi` which can do that automagically for you. You also need to fix the summary of your PR.

Having problem with the web format html tool, it freezes.
How to use the .exe formal tool provided by the game?

**Comment by Proxima-b** [comment]
> Thanks for the electrode source, that seems feasible for a battery electrode at least, though I have my doubts for other applications.

Have have made the change about electrode, what other doubts were you refering?

**Comment by Proxima-b** [comment]
I decide to add duct tapes as a insulation material for dry cells and also the existing disposable cells.

**Comment by anothersimulacrum** [comment]
> > This also highlights that the existing craftable cells need to be revisited, the casing is pretty complex and there's no electrode in the recipe. Ideally you could harvest battery casings and replace/refresh the active ingredients, after looking at their structure I'm increasingly skeptical of from scratch cells that are drop-in replacements for modern batteries.
> > Thanks for the electrode source, that seems feasible for a battery electrode at least, though I have my doubts for other applications.
> 
> The crafting requires blacksmith standard tools with smelting, which is exactly how a complex metal process procedure is represented in dda. If we look closer, many crafting process of the electronic devices in the game are somewhat simplified, like " electrohackl" and "handheld game system", or the UPS, which I'm pretty sure need a dust-free workshop or something else to craft. For lack of a complete industrial system, I think these somewhat simplified recipes do well to balance the gameplay and realism elements of the game. I think the dry battery pieces do not break the overall immersion and game balance. As for the problem of existing craftable cells, they do produce pre-apocalypse batteries, so I think adding the requirement of vaccum molder and polycarbonate sheets would suffice. For now I will try to make a recipe for carbon electrodes.

**Comment by Alex-Folts** [comment]
Dont forget about electrolyte. It called dry cell but still have electrolyte. Where on earth you've gonna get amonium chloride...

---
> #### Purpose of change
> The disposible alkaline batteries currently requires material from the labs to craft which is not always an option early game, especially in a wildness surviving gamethrough with innawood or rural_mapgen mods.

Another solution for original problem would be to add new alkaline battery type, which use sodium hydroxide(lye) electrolyte instead of potassium hydroxide. It should have lower power density and higher internal loses due to lower conductivity of sodium hydroxide, which results in lower effective energy density.

**Comment by Proxima-b** [comment]
> Dont forget about electrolyte. It called dry cell but still have electrolyte. Where on earth you've gonna get amonium chloride...
> 
> > #### Purpose of change
> > The disposible alkaline batteries currently requires material from the labs to craft which is not always an option early game, especially in a wildness surviving gamethrough with innawood or rural_mapgen mods.
> 
> Another solution for original problem would be to add new alkaline battery type, which use sodium hydroxide(lye) electrolyte instead of potassium hydroxide. It should have lower power density and higher internal loses due to lower conductivity of sodium hydroxide, which results in lower effective energy density.

Perhaps it is possible to produce ammonium chloride in game, We've got HCL and ammonia water... But could it potentially make the whole production process way too excruciating?

As for the sodium alkaline battery, it seems legit, but it would be nice to have comparing data from sources so I can determined it's actual energy capacity in game

**Comment by Proxima-b** [comment]
About electrolyte:

Since the electrolyte for Zn-C cell is actually a moisturized mix of manganese dioxide and several acidic chlorids. I decided to introduce a "acidic electrolyte paste" material in the process, instead of using pure manganese dioxide.

In real life, cadmium and ammonium chloride maybe even mercury could be used in the production of dry batteries, but I fear at this point adding more component into this recipe would unnecessarily complicate things.

**Comment by kevingranade** [comment]
> I fear at this point adding more component into this recipe would unnecessarily complicate things.

The goal is representatives, if something requires a lot of different components and is difficult to execute on because of that, the complication is entirely necessary.

**Pull request merged**